external/vcm/vcm/cubedsphere/regridz.py

```diff
@@ -140,25 +164,41 @@ def _regrid_given_delp(
             phalf_fine, ds[var], phalf_coarse_on_fine, z_dim_center=z_dim
         )
 
+    pfull_coarse_on_fine = pressure_at_midpoint_log(
+        delp_coarse_on_fine, dim=z_dim
+    )
     masked_weights = _mask_weights(
-        weights, phalf_coarse_on_fine, phalf_fine, dim_center=z_dim
+        weights,
+        pfull_coarse_on_fine,
+        phalf_coarse_on_fine,
+        phalf_fine,
+        dim_center=z_dim,
+        extrapolate=extrapolate,
     )
 
     return ds_regrid, masked_weights
 
 
 def _mask_weights(
     weights,
+    pfull_coarse_on_fine,
     phalf_coarse_on_fine,
     phalf_fine,
     dim_center=RESTART_Z_CENTER,
     dim_outer=RESTART_Z_OUTER,
+    extrapolate=False,
 ):
-    return weights.where(
-        phalf_coarse_on_fine.isel({dim_outer: slice(1, None)}).variable
-        < phalf_fine.isel({dim_outer: -1}).variable,
-        other=0.0,
-    ).rename({dim_outer: dim_center})
+    if extrapolate:
+        return weights.where(
+            pfull_coarse_on_fine.variable < phalf_fine.isel({dim_outer: -1}).variable,
```

Comment:
I think this is probably "safe" because -1 is always the surface in fv3. But it may be better to make a surface level variable so that it's more clear?
```
surface_level = -1
pfull_coarse_on_fine.variable < phalf_fine.isel({dim_outer: surface_level}).variable
```